data_manager_select_index_by_path/data_manager/test_path_name_value_key_manager.py

```diff
@@ -1,23 +1,121 @@
 #!/usr/bin/env python3
 
-import pytest
 import json
+import os
+import sys
+import tempfile
 from pathlib import Path
 
-from path_name_value_key_manager import DataTable, check_tab
+import pytest
+import yaml
+
+from path_name_value_key_manager import DataTable, check_tab, main
+
+TEST_OUTPUT_DIR = tempfile.mkdtemp(".d",
+                                   "tmp_data_manager_select_index_by_path")
 
 indexes_yml = Path(__file__).parent / Path("indexes.yml")
 test_data = Path(__file__).parent.parent / Path("test-data")
 
 
+def test_application():
+    output_path = tempfile.mkstemp(".json", "fasta_indexes", TEST_OUTPUT_DIR)[
```

Comment:
I would really recommend a fixture here. This is duplicating code that's used in other tests as well, and it also doesn't clean up the temporarily created files after the test finishes. 

Example fixture:

```
@pytest.fixture
def temp_json_path():
    with tempfile.NamedTemporaryFile(suffix=".json", dir=TEST_OUTPUT_DIR) as tmp:
        path =  tmp.name # deletes the actual file as soon as the with block ends
    yield path 
    os.remove(path)   # everything after the yield statement is executed after the test completes
```